lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -909,6 +944,41 @@ static NodeValue simplifyConcatNode(Function *F, ConcatNode *CN) {
     }
   }
 
+  /// If the inputs of concat are all reshape nodes, and the inputs of reshape
+  /// are able to be concatenated, then we can perform concat first and then reshape.
+  /// Concat(Reshape(x) * N) -> Reshape(Concat(x * N)).
+  {
+    std::vector<ReshapeNode *> reshapes;
+    bool optimizable = true;
+    for (auto &I : CN->getInputs()) {
+      if (auto *S = dyn_cast<ReshapeNode>(I)) {
+        reshapes.push_back(S);
+      } else {
+        optimizable = false;
+        break;
+      }
+    }
+    // The inputs of Concat have to be all Reshape nodes, so that we can 
+    // perform the transformation.
+    if (optimizable) {
```

Comment:
LLVM-style prefers early exists to avoid deep nesting and a lot of indentation.
Add e.g.: 
```c++
if(!optimizable) {
  return NodeValue(nullptr);
}
```

Reply:
done